src/logic/basic.lean

```diff
@@ -519,6 +519,17 @@ theorem exists_swap {p : α → β → Prop} : (∃ x y, p x y) ↔ ∃ y x, p x
 @[simp] theorem exists_imp_distrib : ((∃ x, p x) → b) ↔ ∀ x, p x → b :=
 ⟨λ h x hpx, h ⟨x, hpx⟩, λ h ⟨x, hpx⟩, h x hpx⟩
 
+/--
+Extract an element from a existential statement, using `classical.some`.
+-/
+-- This enables projection notation.
+noncomputable abbreviation Exists.some {p : α → Prop} (P : ∃ a, p a) : α := classical.some P
+
+/--
+Show that an element extracted from `P : ∃ a, p a` using `P.some` satisfies `p`.
+-/
+abbreviation Exists.some_spec {p : α → Prop} (P : ∃ a, p a) : p (P.some) := classical.some_spec P
```

Comment:
The linter is complaining that this should be a `lemma`, not a `def`. 

I'm not sure how `abbreviation` , but should `nonempty.some` also be an `abbreviation` for consistency?

Reply:
Why not using a regular def (or a reducible def) instead of abbreviations, which are less standard?

Reply:
I've changed it back to a @[reducible] def. (And changed some_spec to a lemma, which of course it should have been all along.)